packages/components/switch/src/switch.vue

```diff
@@ -109,116 +109,27 @@ import { isBool } from '@element-plus/utils/util'
 import { throwError, debugWarn } from '@element-plus/utils/error'
 import ElIcon from '@element-plus/components/icon'
 import { Loading } from '@element-plus/icons'
-
-import type { PropType, Component } from 'vue'
+import {
+  UPDATE_MODEL_EVENT,
+  CHANGE_EVENT,
+  INPUT_EVENT,
+} from '@element-plus/utils/constants'
+import { switchProps, switchEmits } from './switch'
+import type { SwitchProps } from './switch'
 import type { ElFormContext, ElFormItemContext } from '@element-plus/tokens'
 
-type ValueType = boolean | string | number
-
-interface ISwitchProps {
-  modelValue: ValueType
-  value: ValueType
-  disabled: boolean
-  width: number
-  activeIcon: string | Component
-  inactiveIcon: string | Component
-  activeText: string
-  inactiveText: string
-  activeColor: string
-  inactiveColor: string
-  borderColor: string
-  activeValue: ValueType
-  inactiveValue: ValueType
-  name: string
-  validateEvent: boolean
-  id: string
-  loading: boolean
-  beforeChange?: () => Promise<boolean> | boolean
-}
-
 export default defineComponent({
   name: 'ElSwitch',
   components: { ElIcon, Loading },
-  props: {
-    modelValue: {
-      type: [Boolean, String, Number],
-      default: false,
-    },
-    value: {
-      type: [Boolean, String, Number],
-      default: false,
-    },
-    disabled: {
-      type: Boolean,
-      default: false,
-    },
-    width: {
-      type: Number,
-      default: 40,
-    },
-    inlinePrompt: {
-      type: Boolean,
-      default: false,
-    },
-    activeIcon: {
-      type: [String, Object] as PropType<string | Component>,
-      default: '',
-    },
-    inactiveIcon: {
-      type: [String, Object] as PropType<string | Component>,
-      default: '',
-    },
-    activeText: {
-      type: String,
-      default: '',
-    },
-    inactiveText: {
-      type: String,
-      default: '',
-    },
-    activeColor: {
-      type: String,
-      default: '',
-    },
-    inactiveColor: {
-      type: String,
-      default: '',
-    },
-    borderColor: {
-      type: String,
-      default: '',
-    },
-    activeValue: {
-      type: [Boolean, String, Number],
-      default: true,
-    },
-    inactiveValue: {
-      type: [Boolean, String, Number],
-      default: false,
-    },
-    name: {
-      type: String,
-      default: '',
-    },
-    validateEvent: {
-      type: Boolean,
-      default: true,
-    },
-    id: String,
-    loading: {
-      type: Boolean,
-      default: false,
-    },
-    beforeChange: Function as PropType<() => Promise<boolean> | boolean>,
-  },
-  emits: ['update:modelValue', 'change', 'input'],
-  setup(props: ISwitchProps, ctx) {
+  props: switchProps,
+  emits: switchEmits,
+  setup(props: SwitchProps, ctx) {
```

Comment:
```suggestion
  setup(props, { emit }) {
```

- We needn't add annotation of props
- Avoid using whole `ctx` object